server/actions/ssh/runScript.js

```diff
@@ -1,101 +1,111 @@
+/* eslint-disable prettier/prettier */
 /**
  * This file helps transfer the input.txt needed and then runs the script in the hidden directory
  */
 import { command, transfer } from './';
 import { saveStudents } from '../grade';
 
 // TODO: Need to split this file apart so that we make multiple queries
-export default function runScript(req) {
-  const { assignment, socketId } = req.body;
-  const { repo, input, path, bonusDate, name } = assignment;
-  const repoPath = `.private_repo/${path.match(/.*\/(.+?)\/?$/)[1]}`;
-
-  return command({
-    body: {
-      socketId,
-      command: `rm -r ${repoPath}`
+
+export default async function runScript(req) {
+    const { assignment, socketId } = req.body;
+    const { repo, input, path, bonusDate, name } = assignment;
+    const repoPath = `.private_repo/${path.match(/.*\/(.+?)\/?$/)[1]}`;
+
+    try {
+        await removeProjectDirFromPrivateRepoIfItExist(repoPath, socketId);
+        await transferFileContentToServer(socketId, input, `${repoPath}/input.txt`);
+        await copyFilesToGradeToPrivateRepo(repoPath, path, socketId);
+        const data = await getResultFromGradingScript(repoPath, bonusDate, socketId);
+        const previewList = await getHTMLOutPutFileList(name, socketId);
+
+        if (!data.graders || !previewList) {
+            return Promise.reject({ message: 'Error running script! Please make sure repository path is correct!' });
+        }
+
+        const bonusList = await getBonusList(name, socketId);
+        await saveStudentResults(data, previewList, repo, name, bonusList);
+
+        return data;
+    }catch(err) {
+        console.log(err);
+        return Promise.reject({ message: err });
     }
-  }).then(() => {
-    return transfer({
-      body: {
-        socketId,
-        content: input,
-        filePath: `${repoPath}/input.txt`
-      }
-    }).then(() => {
-      // Copy all files to directory
-      // Copy script to directory
-      // Change permission on script
-      // dos2unix equivalent on script
-      // Run script
-      return command({
+}
+
+async function removeProjectDirFromPrivateRepoIfItExist(repoPath, socketId) {
+    const bashCommand = `pwd;rm -r ${repoPath}`;
+    return executeCommandInShell(socketId, bashCommand);
+}
+
+async function copyFilesToGradeToPrivateRepo(repoPath, path, socketId) {
+    const bashCommand = `cd ${repoPath};pwd;
```

Comment:
Not sure if this was intentional or not but you left a few `pwd` and `ls` throughout the commands.

Reply:
Yea I did that on purpose
I added some code to command.js so that it will print the command and its output stdout/stderr
So I added these so that when other team members are looking at the output they will know where they are and what is in there